src/function/scalar/compressed_materialization/compress_string.cpp

```diff
@@ -0,0 +1,248 @@
+#include "duckdb/common/bswap.hpp"
+#include "duckdb/function/scalar/compressed_materialization_functions.hpp"
+
+namespace duckdb {
+
+static string StringCompressFunctionName(const LogicalType &result_type) {
+	return StringUtil::Format("__internal_compress_string_%s",
+	                          StringUtil::Lower(LogicalTypeIdToString(result_type.id())));
+}
+
+template <class T>
+const uint8_t &FirstByteReference(const T &v) {
+	return reinterpret_cast<const uint8_t *>(&v)[0];
+}
+
+template <class T>
+uint8_t &LastByteReference(T &v) {
+	return reinterpret_cast<uint8_t *>(&v)[sizeof(T) - 1];
+}
+
+template <class T>
+const uint8_t &LastByteReference(const T &v) {
+	return reinterpret_cast<const uint8_t *>(&v)[sizeof(T) - 1];
+}
+
+template <class RESULT_TYPE>
+static inline RESULT_TYPE StringCompressInternal(const string_t &input) {
+	D_ASSERT(input.GetSize() < sizeof(RESULT_TYPE));
+	RESULT_TYPE result;
+	if (sizeof(RESULT_TYPE) <= string_t::INLINE_LENGTH) {
+		memcpy(&result, input.GetPrefix(), sizeof(RESULT_TYPE));
+	} else if (input.GetSize() <= string_t::INLINE_LENGTH) {
+		memcpy(&result, input.GetPrefix(), string_t::INLINE_LENGTH);
```

Comment:
Is this branch necessary/beneficial? `input.GetData()` also includes a branch on `string_t::INLINE_LENGTH`

Reply:
The branching there can be improved. The problem is line 31 (and the return statement) in the regression, though. This happens when compressing a string to a `uint64_t`.

Reply:
I think now, with the `TemplatedReverseMemCpy`, it passes the regression test! I will try to clean up the failing tests and then I think the PR is finally good to go :)

Reply:
Also, the regression test timing has been varying quite a bit, so I think that maybe the hardware that GitHub Actions runs on might vary, giving us a machine that sometimes supports unaligned access better/worse than other machines.